src/devices/Dhtxx/Devices/Dht11.cs

```diff
@@ -24,9 +25,9 @@ public Dht11(int pin, PinNumberingScheme pinNumberingScheme = PinNumberingScheme
         {
         }
 
-        internal override RelativeHumidity GetHumidity(Span<byte> readBuff) => RelativeHumidity.FromPercent(readBuff[0] + readBuff[1] * 0.1);
+        internal override RelativeHumidity GetHumidity(IReadOnlyList<byte> readBuff) => RelativeHumidity.FromPercent(readBuff[0] + readBuff[1] * 0.1);
 
-        internal override Temperature GetTemperature(Span<byte> readBuff)
+        internal override Temperature GetTemperature(IReadOnlyList<byte> readBuff)
```

Comment:
Changing these APIs to take `IReadOnlyList<T>` rather than `Span<T>` will cause a boxing allocation every time `ValueArray<T>` is passed in.

As long as it's within a `fixed` block and the `Span<T>` is not used after the block ends, it should be safe to pass in a `Span<T>` created via `Unsafe.AsPointer(ref _e0)`, as the GC will not move or reclaim the pinned memory during that period.

In the absence of `AsSpan`, that can be achieved like so:
```cs
fixed (byte* p = _readBuff)
{
    device.Read(new Span<byte>(p, _readBuff.Count));
}
```

Reply:
Changed the argument type to directly take `ValueArray<byte>`, so there's no boxing conversion required. The size of this is actually the same as that of a `Span<byte>` so the amount of stack data to copy doesn't change.